app/controllers/application_controller.rb

```diff
@@ -1,15 +1,10 @@
 class ApplicationController < ActionController::Base
   # 全ページをログイン必須とする
   before_action :authenticate_user!
-  before_action :configure_permitted_parameters, only: [:update]
```

Comment:
前回の状態から`, if: :devise_controller?`を付けておけば更新に問題は起きません。
Deviseがここに書くことを推奨しているので，このファイルでよいように思います。

- vendor/bundle/ruby/2.6.0/gems/devise-4.7.1/README.md

>In case you want to permit additional parameters (the lazy way™), you can do so using a simple before filter in your `ApplicationController`:

```ruby
class ApplicationController < ActionController::Base
  before_action :configure_permitted_parameters, if: :devise_controller?

  protected

  def configure_permitted_parameters
    devise_parameter_sanitizer.permit(:sign_up, keys: [:username])
  end
end
```